src/audio/google/google_rtc_audio_processing.c

```diff
@@ -993,34 +1100,66 @@ static int google_rtc_audio_processing_process(struct processing_module *mod,
 		nmax = audio_stream_frames_without_wrap(out_stream, dst);
 		n = MIN(n, nmax);
 		for (i = 0; i < n; i++) {
-			memcpy_s(&(cd->raw_mic_buffer[cd->raw_mic_buffer_frame_index *
-						      cd->num_capture_channels]),
-				 cd->num_frames * cd->num_capture_channels *
-				 sizeof(cd->raw_mic_buffer[0]), src,
-				 sizeof(int16_t) * cd->num_capture_channels);
-			++cd->raw_mic_buffer_frame_index;
-
-			memcpy_s(dst, cd->num_frames * cd->num_capture_channels *
-				 sizeof(cd->output_buffer[0]),
-				 &(cd->output_buffer[cd->output_buffer_frame_index *
-						     cd->num_capture_channels]),
-				 sizeof(int16_t) * cd->num_capture_channels);
-			++cd->output_buffer_frame_index;
-
-			if (cd->raw_mic_buffer_frame_index == cd->num_frames) {
-				GoogleRtcAudioProcessingProcessCapture_int16(cd->state,
-									     cd->raw_mic_buffer,
-									     cd->output_buffer);
-				cd->output_buffer_frame_index = 0;
-				cd->raw_mic_buffer_frame_index = 0;
+			/* If we haven't filled the buffer yet, copy the data */
+			if (cd->raw_mic_buffer_frame_index < cd->num_frames) {
+				size_t num_bytes = sizeof(int16_t) * cd->num_capture_channels;
+				size_t buffer_size = sizeof(cd->raw_mic_buffer);
+				size_t frame_index = cd->raw_mic_buffer_frame_index;
+				size_t buffer_used = frame_index * sizeof(int16_t);
+				size_t buffer_remaining = buffer_size - buffer_used;
```

Comment:
You can combine these equations and eliminate variables those are not used later in this inner loop.